### PR TITLE
Update documentation surrounding contributions

### PR DESCRIPTION
**Pull request by ashao**
Pulls together and creates some new guidelines on how to contribute to SmartSim.

### CODE REVIEW

**Review by juliaputko — APPROVED** [review]
Looks good to me

File: doc/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+1. **It is important to ask questions in the right place.**
+While we understand people's desires for privacy, we strongly prefer the use of
+GitHub issues (feature requests, bug reports or Stack Overflow posts (usage
+questions, best practices) due to their discoverability by future users as
+opposed to private communications like Slack or email. Slack chat should be
+reserved for developer and community discussion. Email is preferred for more
+sensitive items.
+
+1. **Please ask only in one place.** Please restrict yourself to posting your
```
Comment (al-rigazzi) [inline doc/contributing.rst:32]: Should be `2.`

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
One typo to fix, then LGTM! Great stuff!

**Comment by billschereriii** [comment]
Corrected a typo that I saw. Otherwise looks great

**Review by al-rigazzi — APPROVED** [review]
LGTM!

**Review by ankona — APPROVED** [review]

**Pull request merged**